app/static/ringing_room.js

```diff
@@ -1445,41 +1456,57 @@ $(document).ready(function () {
+                // HTTP requests
+                const complib_url_without_http = url_box_contents
+                    .replace("https://", "")
+                    .replace("http://", "")
+                    .toLowerCase();
+                const url_segments = complib_url_without_http.split("/");
+                // Do some basic validation of the URL
+                if (!complib_url_without_http.startsWith("complib.org")) {
+                    this.complib_message = "URL doesn't point to 'complib.org'.";
                     return;
                 }
-                if (!/^\d+$/.test(partial_comp_name)) {
-                    this.current_complib_comp = undefined;
-                    this.complib_error = "Please enter a valid number";
+                if (url_segments.length != 3 || url_segments[1] !== "composition") {
+                    this.complib_message = "URL doesn't point to a composition.";
                     return;
                 }
-                // Keep a reference to the correct 'this'
-                let _this = this;
 
-                let api_url =
-                    "https://api.complib.org/composition/" + partial_comp_name;
-                let standard_url =
-                    "https://complib.org/composition/" + partial_comp_name;
+                /* Send an HTTP request to `complib.org` to figure out what our composition is
+                 * pointing to */
+
+                this.complib_message = "Waiting for CompLib...";
+                let api_url = "https://api." + complib_url_without_http;
+                let standard_url = "https://" + complib_url_without_http;
+
+                let _this = this; // Keep a reference to the correct 'this'
                 $.getJSON(api_url)
-                    .fail(function (_evt, _jqxhr, state) {
+                    .fail(function (evt) {
                         _this.current_complib_comp = undefined;
-                        switch (state) {
-                            case "Bad Request":
-                                _this.complib_error = "Bad request.";
-                                break;
-                            case "Not Found":
-                                _this.complib_error =
-                                    "#" + partial_comp_name + " doesn't exist.";
-                                break;
-                            case "Unauthorized":
-                                _this.complib_error =
-                                    "#" + partial_comp_name + " is private.";
-                                break;
-                            default:
-                                console.warn("Unknown error: " + state);
+                        if (evt.status == 404) {
+                            // Our old favourite, '404 Not found'
+                            _this.complib_message = "Comp #" + complib_url + " doesn't exist.";
```

Comment:
Does this message format make sense with a url instead of an id?

Reply:
Oh, good spot.  And it's an even better spot, because `complib_url` just isn't defined anywhere :sweat_smile:.  I've fixed it now.